### PR TITLE
Improves #1644 - Add an integration test for DELETE /v2/deployments/[id]

### PR DESCRIPTION
**Pull request by gkleiman**

### CODE REVIEW

File: src/test/scala/mesosphere/marathon/integration/AppDeployIntegrationTest.scala [hunk 1]
```diff
@@ -396,6 +396,29 @@ class AppDeployIntegrationTest
     pingTask(taskUpdate2).entityString should be(s"Pong $appId\n")
   }
 
+  test("stop (focefully delete) a deployment") {
```
Comment (aquamatthias) [inline src/test/scala/mesosphere/marathon/integration/AppDeployIntegrationTest.scala]: focefully->forcefully

File: src/test/scala/mesosphere/marathon/integration/AppDeployIntegrationTest.scala [hunk 2]
```diff
@@ -396,6 +396,29 @@ class AppDeployIntegrationTest
...
+    Given("a new app that is not healthy")
+    val appId = testBasePath / "failing"
+    val app = appProxy(appId, "v1", instances = 1, withHealth = false).
```
Comment (aquamatthias) [inline src/test/scala/mesosphere/marathon/integration/AppDeployIntegrationTest.scala]: why do you say withHealth=false and than add the health check manually?
Just use withHealth=true

File: src/test/scala/mesosphere/marathon/integration/AppDeployIntegrationTest.scala [hunk 3]
```diff
@@ -396,6 +396,29 @@ class AppDeployIntegrationTest
...
+      copy(healthChecks = Set(HealthCheck(gracePeriod = 20.second, interval = 1.second, maxConsecutiveFailures = 10)))
+    appProxyCheck(appId, "v1", state = false)
+    val create = marathon.createApp(app)
+    create.code should be (201) // Created
+    val deploymentId = extractDeploymentIds(create).head
+
+    Then("the deployment can not be finished")
+    intercept[AssertionError] {
+      waitForEvent("deployment_success")
```
Comment (aquamatthias) [inline src/test/scala/mesosphere/marathon/integration/AppDeployIntegrationTest.scala]: We always have to wait 30 seconds for this. Is that really necessary? Should id not be enough to check for the (still) running deployment?

**Comment by aquamatthias** [comment]
Just some minor comments. Otherwise looks good.

File: src/test/scala/mesosphere/marathon/integration/AppDeployIntegrationTest.scala (hunk 3 above)
Comment (gkleiman) [inline src/test/scala/mesosphere/marathon/integration/AppDeployIntegrationTest.scala]: You're right, I removed the wait.

File: src/test/scala/mesosphere/marathon/integration/AppDeployIntegrationTest.scala (hunk 2 above)
Comment (gkleiman) [inline src/test/scala/mesosphere/marathon/integration/AppDeployIntegrationTest.scala]: I copied and pasted code that wasn't very nice apparently. I just updated the commit and fixed the test I copied from =).

File: src/test/scala/mesosphere/marathon/integration/AppDeployIntegrationTest.scala (hunk 1 above)
Comment (gkleiman) [inline src/test/scala/mesosphere/marathon/integration/AppDeployIntegrationTest.scala]: :flushed:

**Comment by aquamatthias** [comment]
:+1:  Waiting for the tests to become green - then I will merge.

**Pull request merged**